src/components/Course/CourseCaution.vue

```diff
@@ -38,28 +44,45 @@
 <script lang="ts">
 import { PropType, defineComponent } from 'vue';
 import CourseBaseTooltip from '@/components/Course/CourseBaseTooltip.vue';
-import store from '@/store';
-import { isPlaceholderCourse } from '@/utilities';
+import store, { isCourseConflict } from '@/store';
+import { isPlaceholderCourse, isCourseTaken } from '@/utilities';
 
 type CourseCautions = {
   readonly noMatchedRequirement: boolean;
   readonly typicallyOfferedWarning: readonly string[] | undefined;
   readonly isCourseDuplicate: boolean;
   readonly isPlaceholderWrongSemester: boolean;
+  readonly hasConflictRequirement: boolean;
 };
 
 const getCourseCautions = (
-  course: FirestoreSemesterCourse | FirestoreSemesterPlaceholder,
+  course: FirestoreSemesterCourse | FirestoreSemesterPlaceholder | CourseTaken,
   semesterIndex: number
 ): CourseCautions => {
   const {
     safeRequirementFulfillmentGraph,
     derivedCoursesData: { duplicatedCourseCodeSet, courseToSemesterMap },
   } = store.state;
+
+  const uniqueID = isCourseTaken(course) ? course.uniqueId : course.uniqueID;
+
+  const hasConflictRequirement = !isPlaceholderCourse(course) && isCourseConflict(uniqueID);
+
+  // if a CourseTaken is inputted (thus from the requirements bar), only check for the hasConflictWarning
+  if (isCourseTaken(course)) {
+    return {
+      noMatchedRequirement: false,
+      typicallyOfferedWarning: undefined,
+      isCourseDuplicate: false,
+      isPlaceholderWrongSemester: false,
+      hasConflictRequirement,
+    };
+  }
+
   const noMatchedRequirement =
     !isPlaceholderCourse(course) &&
     safeRequirementFulfillmentGraph.getConnectedRequirementsFromCourse({
-      uniqueId: course.uniqueID,
+      uniqueId: uniqueID,
     }).length === 0;
```

Comment:
I know this is not part of this diff but I just happened to notice this during my AP/IB and constraint violation tasks. I think we should use the dangerous requirement graph here. If the safe requirement graph has no connected requirements from the course, that might mean that all the edges are just in a constraint violation. If the dangerous requirement graph has no connected requirements from the course, that means there is no matched requirement.

Reply:
Pushing this off to a later PR based on you and @einc 's above comments about warnings.

https://www.notion.so/courseplan/Hook-up-FIX-NOW-with-conflict-modal-fix-warning-discrepancy-4922837323ce4476b8a3c4f9caba4b17